cmd/postgres-operator/main.go

```diff
@@ -40,31 +41,66 @@ import (
 	"github.com/crunchydata/postgres-operator/pkg/apis/postgres-operator.crunchydata.com/v1beta1"
 )
 
-// assertNoError panics when err is not nil.
-func assertNoError(err error) {
+// must panics when err is not nil.
+func must(err error) { need(0, err) }
+func need[V any](v V, err error) V {
 	if err != nil {
 		panic(err)
 	}
+	return v
+}
+
+func initClient() (*rest.Config, error) {
+	config, err := runtime.GetConfig()
+
+	if err == nil && userAgent == "" {
+		err = errors.New("call initVersion first")
+	}
+	if err == nil {
+		config.UserAgent = userAgent
+		config.Wrap(otelTransportWrapper())
+
+		// Log Kubernetes API warnings encountered by client-go at a high verbosity.
```

Comment:
Correct me if wrong, but is the magic here that we log at v2 rather than what the native WarningHandler does?

Reply:
Exactly. When unset, controller-runtime calls only `log.Info`:

- https://github.com/kubernetes-sigs/controller-runtime/blob/v0.21.0/pkg/client/client.go#L115
- https://github.com/kubernetes-sigs/controller-runtime/blob/v0.21.0/pkg/log/warning_handler.go#L64